### ISSUE TITLE
Add support for time boundary checks and permissions

### ISSUE THREAD

**Issue by mjf** [influxdata/influxdb#9420]
**TL;DR** (sort of continously evolving feature request, sorry for that)

Hello,

it would be very handy if one could (for backward compatibility)

    > GRANT WRITE ON <database> TO <user>

which would *in fact* be just alias to ``GRANT APPEND`` and ``GRANT DELETE ...`` permissions settings, resulting in something like

    > SHOW GRANTS FOR <user>
    database privileges
    -------- --------------
    database APPEND, DELETE

And then, if one could say

    > REVOKE DELETE ON <database> FROM <user>

it would result in **append-only** permission on the ``<database>`` for the ``<user>`` specified. For instance, in case of a security breach and authentization information leakage the attacker would not be able to alter or delete data in the database, only append new data which represents far less harmful potential. Similarly, one could ``REVOKE APPEND ...`` to have a **delete-only** user permissions on a database which may also be handy in some applications.

Sidenote: Not mentioned above, but of course one would expect to ``GRANT`` both the write permissions separately as well

    > GRANT DELETE ON <database> TO <user>
    > GRANT APPEND ON <database> TO <user>

Sidenote: In fact, I am not sure on what the semantic of write operations should look like in general. From my point of view it **MAY** look like the following scheme:

 - ``WRITE``
   - ``INSERT``
     - *anywhere* (common ``INSERT``)
     - *at "the end" only* (*a.k.a.* ``APPEND``; where the *"the end"* in this case means that the timestamp must not be less or equal to the highest timestamp seen for the specific serie and thus effectively disallowing updates presuming also some mechanism existed to disallow timestamps from the future for insert operation which could represent some **configurable** upper bound for the time window within which insert operation could perform)
   - ``DELETE``
   - ``UPDATE`` (there is no such operation in *InfluxDB* for it makes only a little sense for timeseries database and still having the possibility to make an union of old data with new old data using the common ``INSERT``)

So, this feature request is in fact more about adding **time boundary checks** for insert operation than anything else and should be renamed to *"Add support for time boundary checks and permissions for write operations"* (once having the checks it could apply for ``DELETE`` as well some way). Hm, but what if... having the checks... hm, what if I could limit read operations too? Therefor *"Add support for time boundary checks and permissions"* would probably be the best title variant.

Also if *at least* the actual write permission was split into ``DELETE`` and ``INSERT`` permissions at first it would help a little. User having no ``DELETE`` permission would not at least delete old data, but...

As I told before, I am completely unsure about how to look at the write permissions and what should be done to improve permissions granularity in *InfluxDB*. Thanks.
